### PR TITLE
[Admin][Locale] Index/Create page

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

image
image

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/templates/dashboard/index/content/header/title_block/title.html.twig [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% import '@SyliusAdmin/shared/helper/header.html.twig' as _header %}
 
 <div class="col-12 col-md-6 mb-3 mt-3">
-    <div class="d-md-flex gap-2 align-items-center">
+    <div class="d-md-flex gap-2 align-items-center" {{ sylius_test_html_attribute('dashboard-title') }}>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/templates/dashboard/index/content/header/title_block/title.html.twig]:
```suggestion
    <div class="d-md-flex gap-2 align-items-center" {{ sylius_test_html_attribute('dashboard-header') }}>
```
Along with the element/method names in behat

**Review by NoResponseMate — DISMISSED** [review]

File: src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-use Sylius\Bundle\LocaleBundle\Form\Type\LocaleType;
-use Sylius\Component\Locale\Model\LocaleInterface;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
-use Symfony\Component\Form\AbstractTypeExtension;
-use Symfony\Component\Form\FormBuilderInterface;
-use Symfony\Component\Form\FormEvent;
-use Symfony\Component\Form\FormEvents;
-use Symfony\Component\Intl\Locales;
-
-final class LocaleTypeExtension extends AbstractTypeExtension
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:33]: This extension should be deprecated in 1.14 if we remove it

Comment (Wojdylak) [reply src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:33]: #16539

File: src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php [hunk 3]
```diff
@@ -14,44 +14,52 @@
...
+        private FormElementInterface $formElement,
         private NotificationCheckerInterface $notificationChecker,
-        private LocaleConverterInterface $localeConverter,
     ) {
     }
 
+    /**
+     * @Given I am browsing locales
+     */
+    public function iWantToBrowseLocales(): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]:
```suggestion
    public function iAmBrowsingLocales(): void
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php [hunk 4]
```diff
@@ -121,4 +138,20 @@ public function theLocaleShouldBeStillPresentInTheSystem(string $localeCode): vo
...
+    {
+        Assert::same($this->indexPage->countItems(), 1);
+    }
+
+    /**
+     * @Then I should see the local :localName
+     */
+    public function IShouldSeeTheLocal(string $localName): void
+    {
+        Assert::true($this->indexPage->isSingleResourceOnPage(['name' => $localName]));
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]:
```suggestion
     * @Then I should see the locale :localeName
     */
    public function iShouldSeeTheLocale(string $localeName): void
    {
        Assert::true($this->indexPage->isSingleResourceOnPage(['name' => $localeName]));
```

File: features/admin/locale/managing_locales/filtering_locales.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        And the store has locale "Norwegian (Norway)"
+        And the store has locale "Polish (Poland)"
+        And I am logged in as an administrator
+        And I am browsing locales
+
+    @todo-api @ui
+    Scenario: Filtering locales by code
+        When I filter by code containing "pl"
+        Then I should see a single locale in the list
+        And I should see the local "Polish (Poland)"
```
Comment (GSadee) [inline features/admin/locale/managing_locales/filtering_locales.feature]:
```suggestion
        And I should see the locale "Polish (Poland)"
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/locale/create.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+            sections:
+                template: '@SyliusAdmin/locale/form/sections.html.twig'
+
+        'sylius_admin.locale.create.content.form.sections':
+            general:
+                template: '@SyliusAdmin/locale/form/sections/general.html.twig'
+
+        'sylius_admin.locale.create.content.form.sections.general':
+            code:
+                template: '@SyliusAdmin/locale/form/sections/general/code.html.twig'
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/locale/create.yaml]: I would be here for details:
```suggestion
        'sylius_admin.locale.create.content.form.sections':
            details:
                template: '@SyliusAdmin/locale/form/sections/details.html.twig'

        'sylius_admin.locale.create.content.form.sections.details':
            code:
                template: '@SyliusAdmin/locale/form/sections/general/code.html.twig'
```
but we could make it consistent in a separate PR

**Review by mpysiak — DISMISSED** [review]

**Review by GSadee — DISMISSED** [review]

**Review by mpysiak — APPROVED** [review]

**Review by GSadee — APPROVED** [review]
